### DISCUSSION TITLE
[Feature Request] Extension plugin

### DISCUSSION THREAD

**Discussion by yoyo421** [grafana/grafana#60457] [Plugins]
Currently there are global registry variables in Grafana that store various settings, like colors options for fields and transformers for datasources / fields.

When writing a plugin it is possible to add custom color option in the color registry, the problem with that is when the plugin is not loaded, the custom added color is not loaded, therefor missing. All the current plugins (panel, datasource and app are loading on-demand)

Plugin that loads all time time can bring plugins to a new level of customization, like:
- plugin that adds a bundle of field colors
- plugin that adds a numerous transformers
- plugin that adds more themes to Grafana
 
**Feature request**:
Add "Extension" type plugin that will load on every Grafana page

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
